pkg/snapshotter/snapshotter_csi.go

```diff
@@ -640,6 +655,22 @@ func (c *csiDriver) RestoreVolumeClaim(opts ...Option) (*v1.PersistentVolumeClai
 			}
 
 		}
+		// PB-8482 if restore size is nil then set the pvc size to annotation restoreSize
+		if snapshot.Status.RestoreSize == nil {
+			restoreSizeStr, ok := snapshot.Annotations["px-backup/restoreSize"]
```

Comment:
nit: Since you are using in multiple places i would have preferred to be a constant that can be referenced by all files, like we do for dataExport and others.